src/io/iohk/scalanet/test/peergroup/UDPPeerGroupSpec.scala

```diff
@@ -45,4 +45,27 @@ class UDPPeerGroupSpec extends FlatSpec {
 
     isListeningUDP(pg1.config.bindAddress) shouldBe false
   }
+
+  it should "receive a channel and message that was sent before subscription to the server stream" in withTwoRandomUDPPeerGroups[
+    String
+  ] { (alice, bob) =>
+    val alicesMessage = Random.alphanumeric.take(1024 * 4).mkString
+
+    val aliceClient = alice.client(bob.processAddress).evaluated
+    aliceClient.sendMessage(alicesMessage).runAsync
+
+    expensiveComputation()
```

Comment:
why not `Thread.sleep`, which wouldn't stress CPU needlessly?

Reply:
I noticed that sometimes `Thread.sleep` produced strange behaviours and @loverdos mentioned that its use had also caused flaky tests in some CI tools. The intention was to be illustrative more than a final solution.